### PR TITLE
Document the download endpoint

### PR DESCRIPTION
**Pull request by sereana-fti**

### CODE REVIEW

**Comment by Coldarn** [comment]
Ah nice! Thanks for writing this up! I should have time tomorrow afternoon to review it in depth.

I think I need to add something like `fileDownloadUrl` to the `Ringtail.Context` so this is more discoverable/usable - currently you have to do some manip of the `Ringtail.apiUrl` to compute the download URLs which isn't clear or documented anywhere, hah.

File: DownloadFiles.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Download Document Files
+Extensions can download the files associated with documents.
+
+A UIX can identify the file to download in the following ways:
+- Download a file based on the file ID (9.6.002)
+- Download a content file based on the content file type ranking (9.6.002)
+- Download a page file based on the page file number (9.6.002)
+- Download a page file as a .jpg or .png image (9.7.000)
```
Comment (Coldarn) [inline DownloadFiles.md]: Minor point, the image download API downloads images of pages, not page files. Page "files" can be PDFs with many pages inside, but the image download API would fetch images of just one of it's pages.

File: DownloadFiles.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+## Usage notes
+Note the following:
+- This endpoint allows users to download files only through a UIX. Users cannot download files by calling the endpoint directly.
+- A user can download a file though a UIX even if the security for the **Document - Download** option on the Case Home > Security > Features page in Ringtail is set to Deny for the user's group.
+- A user cannot download a file associated with a document if the user does not have access to the document, or if the document contains branded redactions.
+- Images are returned at 100% scale. This may result in large images being returned, particularly for documents that were scanned at a high DPI (dots per inch).
```
Comment (Coldarn) [inline DownloadFiles.md]: Also, I think the image download API only works for documents that have been imaged (i.e. in PDF or some image format, not for native files like word docs).

File: DownloadFiles.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+## Request format
+### Endpoint
+`POST /api/download`
```
Comment (Coldarn) [inline DownloadFiles.md]: The URL will actually be something like `/Ringtail-Svc-Portal/api/download`, but unfortunately the first segment of the path varies based on how the app is installed.

The most reliable way to get the URL is to take `Ringtail.Context.apiUrl` and replace `query` with `download`. Not pretty, need a better solution for this.

File: DownloadFiles.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+Depending on how the UIX identifies the file to download, the required parameters vary.
+
+**To download a file based on the file ID:**
+
+| Parameter | Description | Type | Required? | Notes |
+| --- | --- | --- | --- | --- |
+| caseId | Case ID. Identifies the case in the portal. | int | Required |  |
+| fileId | File ID. Identifies the file in the case. | string | Required |  |
+
+For example: http://ringtail.example.com/Ringtail-Svc-Portal/api/download?caseId=6078&fileId=29384
```
Comment (Coldarn) [inline DownloadFiles.md]: `fileId`s look more like `C.12345` or `P.67890`. Is it worth mentioning that file IDs come from the API? We don't have a place to link them to I don't think.

File: DownloadFiles.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+For example: http://ringtail.example.com/Ringtail-Svc-Portal/api/download?caseId=3024&mainId=39456&contentFileRank=1
+
+**To download a page file based on the page file number:**
+
+| Parameter | Description | Type | Required? | Notes |
+| --- | --- | --- | --- | --- |
+| caseId | Case ID. Identifies the case in the portal. | int | Required |  |
+| mainId | Document Main ID. Identifies the document in the case. | int | Required |  |
+| pageFileNumber | Sequential order of the file among the document's page files. | int | Required |  |
```
Comment (Coldarn) [inline DownloadFiles.md]: I believe pageFileNumbers also use 1-based indexing, so the first page file is 1, then 2, etc.

File: DownloadFiles.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+| Parameter | Description | Type | Required? | Notes |
+| --- | --- | --- | --- | --- |
+| caseId | Case ID. Identifies the case in the portal. | int | Required |  |
+| mainId | Document Main ID. Identifies the document in the case. | int | Required |  |
+| pageId | Page ID. Identifies the page file in the case. | int | Required |  |
+| pageNum | The number of the page that you want to return, for a page file that has multiple pages. | int | Required |  |
+| imageFormat | The file type that you want to return the image as. Specify one of `jpg` or `png`. | string | Optional | Default is jpg. |
+
+For example: http://ringtail.example.com/Ringtail-Svc-Portal/api/download?caseId=201&mainId=29384&pageId=1&pageNum=4&pageId=1&imageFormat=png
```
Comment (Coldarn) [inline DownloadFiles.md]: `pageId` will be larger numbers, like 23901. Using an example of 1 is more misleading IMO.

File: DownloadFiles.md (hunk 1 above)
Comment (sereana-fti) [reply DownloadFiles.md]: I fixed this and the other stuff. Have another look.

**Pull request merged**